src/sentry/static/sentry/app/utils/apm.jsx

```diff
@@ -30,12 +30,72 @@ function startTransaction() {
   finishTransaction(5000);
 }
 
+const requests = new Map([]);
+const renders = new Map([]);
 let flushTransactionTimeout = undefined;
-function finishTransaction(delay) {
+let interruptFlush = false;
+export function finishTransaction(delay) {
   if (flushTransactionTimeout) {
     clearTimeout(flushTransactionTimeout);
   }
-  flushTransactionTimeout = setTimeout(() => Sentry.finishSpan(), delay || 5000);
+  if (
+    Array.from(requests).find(([, active]) => active) ||
```

Comment:
This could be moved to something like
```js
function hasActiveMember(dict) {
  return Array.from(requests).find(([, active]) => active)
}

// then

if (hasActiveMember(requests) || hasActiveMember(renders)) { ... }
```

as now it's duplicated in 4 places.